executor/src/witgen/jit/processor.rs

```diff
@@ -140,39 +141,62 @@ impl<'a, T: FieldElement, FixedEval: FixedEvaluator<T>> Processor<'a, T, FixedEv
 
         // TODO Tuning: If this fails (or also if it does not generate progress right away),
         // we could also choose a different variable to branch on.
-        let left_branch_code =
-            self.generate_code_for_branch(can_process.clone(), first_branch, branch_depth + 1)?;
-        let right_branch_code =
-            self.generate_code_for_branch(can_process, second_branch, branch_depth + 1)?;
-        let code = if left_branch_code == right_branch_code {
-            common_code.into_iter().chain(left_branch_code).collect()
-        } else {
-            common_code
+
+        let first_branch_result =
+            self.generate_code_for_branch(can_process.clone(), first_branch, branch_depth + 1);
```

Comment:
```suggestion

        enum RecoverableResult<T: FieldElement> {
            Code(Vec<Effect<T, Variable>>),
            ConstraintUnsatisfiable,
        }

        let first_branch_result = self
            .generate_code_for_branch(can_process.clone(), first_branch, branch_depth + 1)
            .map(RecoverableResult::Code)
            .map_err(|e| {
                if e.reason == ErrorReason::ConflictingConstraints {
                    Ok(RecoverableResult::ConstraintUnsatisfiable)
                } else {
                    Err(e)
                }
            })?;
```

What do you think of something like this to simplify the cases below?